### ISSUE TITLE
TNC assets should be behind Cloudfront

### ISSUE THREAD

**Issue by TylerFisher** [news-catalyst/next-tinynewsdemo#687]
Right now we're hitting our image files directly on S3. They should be behind a Cloudfront CDN. We should host this CDN at a subdomain on tinynewsco.org. Maybe assets.tinynewsco.org?

https://aws.amazon.com/premiumsupport/knowledge-center/cloudfront-serve-static-website/

**Comment by jacqui** [comment]
I've started working on this - where is tinynewsco.org registered? I don't see it in Route53/AWS. If we want to serve assets behind a cloudfront distribution at `assets.tinynewsco.org` the instructions (https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/distribution-web-values-specify.html#DownloadDistValuesCNAME) include setting up CNAMEs...

**Comment by jacqui** [comment]
FYI Cloudfront Distribution fronting tnc-test-uploads bucket: https://console.aws.amazon.com/cloudfront/v3/home?region=us-east-1#/distributions/E15QK0I2XFMOZ

**Comment by jacqui** [comment]
To clarify, we will need to add a CNAME mapping `assets.tinynewsco.org` to `d2ptro4efyua2o.cloudfront.net`. Once we do that, we can update the image hostname/domain in the front-end app.

**Comment by TylerFisher** [comment]
Aron owns it. I'll work with him to get the CNAME set up.

**Comment by TylerFisher** [comment]
Okay, the CNAME is set, but I get 403s when I try to load an image: https://assets.tinynewsco.org/oaklyn/null/imagekix.rokfk7nr0sud.png

**Comment by jacqui** [comment]
@TylerFisher you're right, once the CNAME was setup I still had to complete a couple of steps to configure cloudfront. 

tldr: do you have the SSL cert? looks like we have to upload it in order to finish configuring cloudfront. You upload it in the [edit settings](https://console.aws.amazon.com/cloudfront/v3/home?region=us-east-1#/distributions/E15QK0I2XFMOZ/edit) for the CF distribution. Docs are linked below.

<img width="841" alt="Screen Shot 2021-07-26 at 8 41 10 am" src="https://user-images.githubusercontent.com/3397/126915647-2ef274a5-c097-4bd1-be8b-29e5b2cc5eca.png">

1. [Edit the distribution](https://console.aws.amazon.com/cloudfront/v3/home?region=us-east-1#/distributions/E15QK0I2XFMOZ/edit)
2. Enter `assets.tinynewsco.org` as an "Alternate Domain Name"
3. Apparently we also need to upload the security certificate in order to specify an Alternate Domain Name as the edit form won't submit without it 😫

<img width="859" alt="Screen Shot 2021-07-26 at 8 43 10 am" src="https://user-images.githubusercontent.com/3397/126915681-a3ff45e4-9509-4f46-9706-ab86e166d8d7.png">

[documentation on alternate domain name requirements mentioned in screenshot](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/CNAMEs.html#alternate-domain-names-requirements)

```
To add an alternate domain name (CNAME) to a CloudFront distribution, you must attach to your distribution a trusted, valid SSL/TLS certificate that covers the alternate domain name. This ensures that only people with access to your domain’s certificate can associate with CloudFront a CNAME related to your domain. 
```
